Trackers/iou_pred_tracker/iou_pred.py

```diff
@@ -0,0 +1,222 @@
+        history_sum = list(np.sum(history,axis=0))
+        h1 = history_sum[0]/3
+        h2 = history_sum[1]/3
+        h3 = history_sum[2]/3
+        h4 = history_sum[3]/3
+        area_ratio = abs(((h3-h1) * (h4-h2)) / ((bbox[2]-bbox[0]) * (bbox[3]-bbox[1])))
+        bbox = list(bbox)
+        bbox[0], bbox[1], bbox[2], bbox[3] = (bbox[0] + bbox[2])/2, (bbox[1] + bbox[3])/2, bbox[2] - bbox[0], bbox[3] - bbox[1]
+        h1, h2, h3, h4 = (h1 + h3)/2, (h2 + h4)/2, h3 - h1, h4 - h2
+
+        x_offset = bbox[0] - h1
+        y_offset = bbox[1] - h2
+
+        predicted_bbox = [bbox[0] + x_offset, bbox[1] + y_offset, bbox[2] * math.sqrt(area_ratio), bbox[3] * math.sqrt(area_ratio)]
+        predicted_bbox = [predicted_bbox[0] - predicted_bbox[2]// 2, predicted_bbox[1] - predicted_bbox[3]// 2, predicted_bbox[0] + predicted_bbox[2]// 2, predicted_bbox[1] + predicted_bbox[3]// 2]
+        
+     
+        return tuple(predicted_bbox)
+
+class iou_pred_tracker():
+    def __init__(self):
+        self.tracks = []
+        self.nextID = 0
+        self.max_age = 50
+        self.min_hits = 3
+
+    def add_track(self, id, bbox):
+        """
+        Adding newly tracked objects  to track list.
+        
+        Args
+        id : new id to be assigned 
+        bbox : bounding box of the object
+        centroid : centroid of the object
+
+        Returns
+        None
+        """
+        
+        track = iou_pred_track(id, bbox , 1, 0)
+        self.tracks.append(track)
+        self.nextID += 1
+        print("added id ",track.id, "succesfully")
+
+    def delete_track(self):
+        """
+        Deletes a centroid track object from tracks list if the max age is crossed.
+        
+        Args
+        None
+
+        Returns 
+        None
+        """
+
+        for track in self.tracks:
+            if(track.miss > self.max_age):
+                print("deleted id ",track.id, "succesfully")
+                self.tracks.remove(track)
+    
+    def update(self, detections):
+        """
+        Returns the active list of centroid track objects
+
+        Args
+        detections : list of bbox of detected objects
+
+        Returns
+        result : list of objects of class Track
+        """
+
+        if(detections == []):
+            if(self.tracks == []):
+                return self.tracks
+            else:
+                # increase every objects miss by 1
+                for track in self.tracks:
+                    track.miss +=1
+                self.delete_track()
+                return self.tracks
+        if len(self.tracks) == 0:
+            for i in range(0, len(detections)):
+                self.add_track(self.nextID, detections[i])
+            return self.tracks
+        tracks = []
+        predicts = []
+        for trk in self.tracks:
+            tracks.append(trk.bbox)
+            if(trk.hits > self.min_hits):
+                predict_bbox = trk.iou_predict(trk.bbox, trk.history)
```

Comment:
What's the logic behind passing the arguements to the method of the same class object which already have them.